controllers/util/storageconsumer.go

```diff
@@ -305,3 +307,77 @@ func GetStorageConsumerDefaultResourceNames(
+	rbdProvisionerSecretName := storageClaimCephCsiSecretName("provisioner", rbdStorageRequestHash)
+	rbdStorageRequestName := GetStorageRequestName(storageConsumerUID, rbdClaimName)
+	rbdStorageRequestMd5Sum := md5.Sum([]byte(rbdStorageRequestName))
+	rnsName := fmt.Sprintf("cephradosnamespace-%s", hex.EncodeToString(rbdStorageRequestMd5Sum[:16]))
+
+	// For Provider Mode we supported creating one cephFs claim where we generate clientProfile, secret and svg name using
+	// consumer UID and Storage Claim name
+	// The name of StorageClass is the same as name of storageClaim in provider mode
+	cephFsClaimName := GenerateNameForCephFilesystemStorageClass(storageCluster)
+	cephFsClaimMd5Sum := md5.Sum([]byte(cephFsClaimName))
+	cephFSClientProfile := hex.EncodeToString(cephFsClaimMd5Sum[:])
+	cephFsStorageRequestHash := getStorageRequestHash(storageConsumerUID, cephFsClaimName)
+	cephFsNodeSecretName := storageClaimCephCsiSecretName("node", cephFsStorageRequestHash)
+	cephFsProvisionerSecretName := storageClaimCephCsiSecretName("provisioner", cephFsStorageRequestHash)
+	cephFsStorageRequestName := GetStorageRequestName(storageConsumerUID, cephFsClaimName)
+	cephFsStorageRequestMd5Sum := md5.Sum([]byte(cephFsStorageRequestName))
+	svgName := fmt.Sprintf("cephfilesystemsubvolumegroup-%s", hex.EncodeToString(cephFsStorageRequestMd5Sum[:16]))
+
+	resourceMap.ReplaceRbdRadosNamespaceName(rnsName)
+	resourceMap.ReplaceSubVolumeGroupName(svgName)
+	resourceMap.ReplaceSubVolumeGroupRadosNamespaceName("csi")
+	resourceMap.ReplaceRbdClientProfileName(rbdClientProfile)
+	resourceMap.ReplaceCephFsClientProfileName(cephFSClientProfile)
+	resourceMap.ReplaceCsiRbdNodeCephUserName(rbdNodeSecretName)
+	resourceMap.ReplaceCsiRbdProvisionerCephUserName(rbdProvisionerSecretName)
+	resourceMap.ReplaceCsiCephFsNodeCephUserName(cephFsNodeSecretName)
+	resourceMap.ReplaceCsiCephFsProvisionerCephUserName(cephFsProvisionerSecretName)
+
+}
+
+// GetStorageRequestName generates a name for a StorageRequest resource.
+func GetStorageRequestName(consumerUUID, storageClaimName string) string {
+	return fmt.Sprintf("storagerequest-%s", getStorageRequestHash(consumerUUID, storageClaimName))
+}
+
+// getStorageRequestHash generates a hash for a StorageRequest based
+// on the MD5 hash of the StorageClaim name and storageConsumer UUID.
+func getStorageRequestHash(consumerUUID, storageClaimName string) string {
+	s := struct {
+		StorageConsumerUUID string `json:"storageConsumerUUID"`
+		StorageClaimName    string `json:"storageClaimName"`
+	}{
+		consumerUUID,
+		storageClaimName,
+	}
+
+	requestName, err := json.Marshal(s)
+	if err != nil {
+		panic("failed to marshal storage class request name")
+	}
+	md5Sum := md5.Sum(requestName)
+	return hex.EncodeToString(md5Sum[:16])
+}
+
+func storageClaimCephCsiSecretName(secretType, suffix string) string {
+	return fmt.Sprintf("ceph-client-%s-%s", secretType, suffix)
+}
```

Comment:
These does not seems like they belong here

Reply:
These functions are used for configmap value generation. The reason I chose this place was that this file contains the interface for consumer configmap. I can move it to k8sutils or a new file. What do you suggest